### ISSUE TITLE
MatterBridge Shelly plugin not adding devices

### ISSUE THREAD

**Issue by rlhead** [Luligu/matterbridge-shelly#100]
I'm not sure what I'm missing in my setup. I have MatterBridge installed and the Shelly plugin installed and it all appears to be running fine. I can see that it is locating and communicating with the shelly devices on my network, however, not one shelly devices gets added into the devices under the Shelly plugin. I have a lot of shelly devices (about 30 or so). Many of them (gen 1 devices) are running a mongoose HomeKit firmware, so I'm not expecting to see those devices through this plugin, however I have some dimmer's (gen 1 and gen 2) that are running the native and current shelly firmware, and although I see them in the log, they don't ever show up as "devices" under this plugin, nor do they show up in HomeKit via the MatterBridge. I've tried clearing the discovery cache/storage several times with no luck.

I've attached some log files. shellydimmer-D0DDD6 host xxx.xxx.xxx.205 is an example of a gen1 Dimmer running the latest Shelly firmware. 
Device shellypluswdus-441793CCB144 is a gen 2 device and in the logs it says it was discovered and added, but the plugin still shows 0 devices, and it doesn't show up in HomeKit. 

Just not sure if I'm missing some config or if it's a problem with the over all setup. Any help is greatly appreciated. Thank you!

matterbridge v. 1.6.5
Shelly plugin v. 1.0.11
[matter-2.log](https://github.com/user-attachments/files/18046446/matter-2.log)
[matterbridge-2.log](https://github.com/user-attachments/files/18046447/matterbridge-2.log)

**Comment by Luligu** [comment]
Hi,
first of all thanks for your interest.

First issue: the unofficial firmware mongoose is of course not supported and also useless since you can pair matterbridge with all controllers (Home included) and creates a lot of issues. I can try to remove all of them from mdns scanner so they don't even pop up.

Second issue [21:42:13.426] [Matterbridge shelly plugin] [error] Shelly device shellyix3-E8DB84D21853 host 192.168.128.202 failed to register with Matterbridge: Error: Only supported EndpointInterface implementation is Endpoint 
this is very strange and happened only with a plugin of tb0str that never worked and will never work.

To understand why you have this problem I need to know exactly how and where you installed matterbridge. 
I see from the log that you used nvm: that is not a good choice cause nvm is for debugging and development. I would remove it and install node from nodesource https://github.com/nodesource/distributions?tab=readme-ov-file#using-debian-as-root-nodejs-22.

**Comment by rlhead** [comment]
Thanks for the quick reply!

Agree, about the unofficial firmware. I wasn't expecting them to show up, and was planning to move them back to Shelly firmware once I have this Matterbridge shelly plugin operating as it should.

Thanks for the tip about NVM. I was setting this all up on a virtual machine to test the config out and plan to install this all on a Raspberry PI. I won't use NVM to install node in that environment, per your suggestion.

After posting my initial post, I thought I'd try uninstalling the Shelly plugin, reboot the system and re-install the plugin, and to my surprise, and relief, the 4 devices I was expecting to show up, where discovered and added as devices. They all work as expected. So happy to see this, as now I can work to migrate all shelly to this matterbridge setup.
That Shelly i3 device at x.x.x.202 did end up registering and showing up after the reinstall of the Shelly plugin.

Here is how I went about installing. Setup Debian in a VM, added NVM and then added node. Installed Matterbridge, started that, and attempted to install the Shelly plugin via the frontend of Matterbridge, but that failed. So I installed it via CLI and that was successful, so I didn't think too much about that. When I removed and then reinstalled the plugin, I did this all through the frontend web interface and it was successful. 

It feels like the failed plugin install (initial attempt) via the front end may have created some issue. Sorry I don't have any details or logs from that.

I really appreciate you taking a look at the logs and I'm really looking forward to using Matterbridge for all of my shelly devices. Thank you so much for working on this project!!

Thanks!
